src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/multiblocks/ArmorForge.java

```diff
@@ -71,26 +66,51 @@ private boolean isCraftable(Inventory inv, ItemStack[] recipe) {
         return true;
     }
 
-    private void craft(Player p, ItemStack output, Inventory inv, Inventory outputInv) {
-        for (int j = 0; j < 9; j++) {
-            ItemStack item = inv.getContents()[j];
+    private Inventory createVirtualInventory(Inventory inv) {
+        Inventory fakeInv = Bukkit.createInventory(null, 9, "Fake Inventory");
+
+        for (int j = 0; j < inv.getContents().length; j++) {
+            ItemStack stack = inv.getContents()[j];
 
-            if (item != null && item.getType() != Material.AIR) {
-                ItemUtils.consumeItem(item, true);
+            if (stack != null) {
+                stack = stack.clone();
+                ItemUtils.consumeItem(stack, true);
             }
+
+            fakeInv.setItem(j, stack);
         }
 
-        for (int j = 0; j < 4; j++) {
-            int current = j;
+        return fakeInv;
+    }
```

Comment:
Is this method copied from anywhere?
I think it would make sense to make it a static utility method or so since we would otherwise face a lot of redundancies and duplications.

Reply:
> Is this method copied from anywhere?

Yes, from BackpackCrafter: https://github.com/Slimefun/Slimefun4/blob/c9adba2f39068490518bafe0cb8fd9bc04a7b572/src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/multiblocks/BackpackCrafter.java#L43

How do I tackle this? Make `ArmorForge extends BackpackCrafter` instead of MultiBlockMachine since latter is already extended in BackpackCrafter. Sorry. I know its messy but I am learning and loving it so far :D

Reply:
I think making it extend BackpackCrafter should be fine, even though the BackpackCrafter class might need a new name after this XD